# README 

Descriptions of files and code for analyzing plant transcriptomes. 

### LPWG_2018_data.csv 
Trait data for all plant species in legume phylogeny from the Legume Phylogeny Working Group 2013 Taxon paper

### LPWG_2018_newick_spp.txt 
Phylogeny of legume species from the Legume Phylogeny Working Group 2013 Taxon paper (pruned for species with trait data) 

### MappedRoyGenes.csv 
Genes in plant transcriptomes that mapped to symbiosis genes identified in Roy et al. 2020 Plant Cell paper

### Plant_PAML_results_NEW.csv 
dN/dS ratios for separate genes in each plant species in the datast 

### Plant_rates.Rmd 
R script for analyzing dN/dS ratios for genome-wide genes, symbiosis genes, and invasion data in mutualistic and non-mutualistic plant species 

### Plant_rates.md 
R script for analyzing dN/dS ratios for genome-wide genes, symbiosis genes, and invasion data in mutualistic and non-mutualistic plant species 

### TwoRateCombo_clean.csv 
dN/dS ratios from two-rate models performed in PAML for plant species 

### dnds_workflow_plants.txt 
Code for analyzing raw RNA-seq reads to obtain dN/dS ratios genome-wide, at symbiosis genes, and using two-rate models in PAML 

### invasionrates_new.csv 
Datasheet for additional analyses comparing plant species that are invasive and non-invasive 
