# README 

Description of datasheets and code for analyzing bacteria and Ensifer genomes 

### EniSamples.csv
Accession codes for genomes used in the Ensifer genus analysis 

### EniSamplesClean.csv 
Trait data for genomes used in the Ensifer genus analysis 

### Fagorzi_data2.csv 
Extended trait data for all genomes used in the Ensifer genus analysis including the assembly and accession codes 
Symbiotic status of the genomes coded in the column named "TIA_NGENES" 

### Proteo_nodule.csv
Species across the Proteobacteria phylogeny used in the analysis 
Symbiotic status of the genomes coded in the column "Nodulating"

### Proteo_tree.txt 
Phylogeny of Proteobacteria species from Rahimlou et al. 2021 paper 

### Rhizobia_pair2NEW_results.csv 
dN/dS ratio data from the paired analysis of symbiotic and non-symbiotic bacteria across Proteobacteria phylogeny 

### Rhizobia_rates.Rmd 
R code for all analyses of paired bacteria species, Ensifer genus, and chromsome/pSymB analyses 
Running this code with all files in this directory will produce the results presented in the paper 

### Rhizobia_rates.md 
R code for all analyses of paired bacteria species, Ensifer genus, and chromsome/pSymB analyses 

### dNdS_workflow_bacteria.txt
All code for analyzing the raw genome data of the paired symbiotic and non-symbiotic bacteria species 
Output will be dN/dS ratios for each gene for each species in the analysis 

### dNdS_workflow_ensifer.txt 
All code for analyzing the raw genome data of the Ensifer genus species 
Output will be dN/dS ratios for the symbiotic and non-symbiotic clade in the Ensifer genus

### psymBorthosMEL.csv 
Orthologs on pSymB identified using the S. meliloti species as the reference 

### sinotreefinal.csv
Trait data for species in the Ensifer genus analysis 

### symbiotic_orthos.csv 
Orthologs identified as symbiotic genes using key word (ex. nod, nif, etc) searches in annotated genomes 

### unrootedPhylogenyEnsifer.tre 
Phylogeny of species in the Ensifer genus including symbiotic and non-symbiotic species from Fagorzi et al 2020 Genome BioL Evol paper 

